src/pages/home/report/ReportActionItemParentAction.tsx

```diff
@@ -1,91 +1,105 @@
+                    >
+                        <ReportActionItem
+                            // @ts-expect-error TODO: Remove this once ReportActionItem (https://github.com/Expensify/App/issues/31982) is migrated to TypeScript.
+                            onPress={() => Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(ancestor.report.reportID))}
+                            report={ancestor.report}
+                            action={ancestor.reportAction}
+                            displayAsGroup={false}
+                            isMostRecentIOUReportAction={false}
+                            shouldDisplayNewMarker={ancestor.shouldDisplayNewMarker}
+                            index={index}
+                        />
+                        {!ancestor.shouldHideThreadDividerLine && <View style={[styles.threadDividerLine]} />}
+                    </OfflineWithFeedback>
+                ))}
             </View>
-            {!shouldHideThreadDividerLine && <View style={[styles.threadDividerLine]} />}
-        </OfflineWithFeedback>
+        </>
     );
 }
 
 ReportActionItemParentAction.displayName = 'ReportActionItemParentAction';
 
 export default withOnyx<ReportActionItemParentActionProps, ReportActionItemParentActionOnyxProps>({
-    report: {
-        key: ({reportID}) => `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+    // We should subscribe all reports and report actions here to dynamic update when any parent report action is changed
+    allReportActions: {
+        key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
     },
-    parentReportActions: {
-        key: ({parentReportID}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
-        canEvict: false,
+    allReports: {
+        key: ONYXKEYS.COLLECTION.REPORT,
     },
-})(ReportActionItemParentAction);
+})(
+    memo(ReportActionItemParentAction, (prevProps, nextProps) => {
+        const {allReportActions: prevAllReportActions, allReports: prevAllReports, ...prevPropsWithoutReportActionsAndReports} = prevProps;
+        const {allReportActions: nextAllReportActions, allReports: nextAllReports, ...nextPropsWithoutReportActionsAndReports} = nextProps;
+
+        const prevReport = prevAllReports?.[`${ONYXKEYS.COLLECTION.REPORT}${prevProps.reportID}`];
+        const nextReport = nextAllReports?.[`${ONYXKEYS.COLLECTION.REPORT}${nextProps.reportID}`];
+        const prevAllAncestors = ReportUtils.getAllAncestorReportActions(prevReport, prevProps.shouldHideThreadDividerLine ?? false, prevAllReports, prevAllReportActions);
+        const nextAllAncestors = ReportUtils.getAllAncestorReportActions(nextReport, nextProps.shouldHideThreadDividerLine ?? false, nextAllReports, nextAllReportActions);
+
+        if (prevReport !== nextReport || !deepEqual(prevAllAncestors, nextAllAncestors)) {
+            return false;
+        }
+
+        return lodashIsEqual(prevPropsWithoutReportActionsAndReports, nextPropsWithoutReportActionsAndReports);
+    }),
```

Comment:
Ok, so this is what we are doing to address the performance issues of subscribing to all the reports and report actions.

My gut is saying this looks a bit hacky and will hurt performance. 

I would think it would be better to do something like:

- Only do this kind of subscription if the report action would ever have an ancestor (which knocks out a huge chunk of report actions)
- If we merge a report or report action into Onyx rebuild the ancestors for the report action
- Cache that somewhere and let all the child report actions know about the update

That's just one idea though - I am not suggesting that this is what you _have to_ do. Maybe we can discuss other ideas before proceeding further.